degirum_face/discovery.py

```
@@ -0,0 +1,148 @@
+2. Compatible hardware (intersection of registry and system)
+3. Compatible models based on hardware/task filters
+
+Note: For registry hardware/tasks, use degirum_face.model_registry.get_hardware()
+and degirum_face.model_registry.get_tasks() directly.
+"""
+
+from typing import List, Dict, Any, Optional
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def get_system_hw(inference_host_address: str = "@local") -> List[str]:
+    """
+    Get hardware available on a specific inference host
+
+    Args:
+        inference_host_address (str): Where to check for hardware
+            - "@local": Local system using degirum.get_supported_devices()
+            - "@cloud": DeGirum cloud using degirum.get_supported_devices()
+            - "host_ip": AI server using degirum.get_supported_devices()
+
+    Returns:
+        List[str]: List of available hardware types (e.g., ['HAILORT/HAILO8', 'TFLITE/CPU'])
+                   Returns empty list if detection fails
+    """
+    try:
+        import degirum as dg
+
+        logger.debug(f"Getting supported devices from {inference_host_address}")
+        supported_devices = dg.get_supported_devices(
+            inference_host_address, "degirum/public"
+        )
+
+        # Extract hardware types from supported devices
+        hardware_list = []
+
+        for device in supported_devices:
+            # Device format might be like "HAILORT/HAILO8", "TFLITE/CPU", etc.
+            if isinstance(device, str):
+                hardware_list.append(device)
+            elif isinstance(device, dict) and "name" in device:
+                hardware_list.append(device["name"])
+
+        hardware_list = sorted(list(set(hardware_list)))
+        logger.info(f"Detected hardware on {inference_host_address}: {hardware_list}")
+        return hardware_list
+
+    except ImportError:
+        logger.error("degirum package not available for hardware discovery")
+        return []
+    except Exception as e:
```

Generally, it is not a good practice to hide exceptions. 
Proposition: just leave the following;
```
def get_system_hw(inference_host_address: str = "@local") -> List[str]:
    return sorted(dg.get_supported_devices(inference_host_address, "degirum/public")) 
```